onnxscript/function_libs/torch_aten/ops/core.py

```diff
@@ -4208,19 +4208,70 @@ def aten_native_dropout_backward(
     raise NotImplementedError()
 
 
+@torch_op("aten::native_group_norm", trace_only=True)
 def aten_native_group_norm(
-    input: TensorType,
-    weight: Optional[TensorType],
-    bias: Optional[TensorType],
-    N: INT64,
-    C: INT64,
-    HxW: INT64,
-    group: int,
-    eps: float,
-) -> tuple[TensorType, TensorType, TensorType]:
+    input: TFloat,
+    weight: Optional[TFloat],
+    bias: Optional[TFloat],
+    N: Optional[INT64] = None,  # pylint: disable=unused-argument
+    C: Optional[INT64] = None,  # pylint: disable=unused-argument
+    HxW: Optional[INT64] = None,  # pylint: disable=unused-argument
+    group: Optional[int] = None,
+    eps: Optional[float] = 1e-05,
+) -> Tuple[TFloat, TFloat, TFloat]:
     """native_group_norm(Tensor input, Tensor? weight, Tensor? bias, SymInt N, SymInt C, SymInt HxW, int group, float eps) -> (Tensor, Tensor, Tensor)"""
 
-    raise NotImplementedError()
+    # Actually we don't need N,C,HxW value because the input tensor has that information
+    if group is None:
+        group = 1  # Equal to LayerNorm
+
+    if weight is None:  # Set to 1.0 as default, the shape is Channel size
+        weight = op.Expand(op.Constant(value_floats=[1.0]), op.Shape(input, start=1, end=2))
+
+    if bias is None:  # Set to 0.0 as default, the shape is Channel size
+        bias = op.Expand(op.Constant(value_floats=[0.0]), op.Shape(input, start=1, end=2))
+
+    norm, fake_mean, fake_var = _aten_native_group_norm_onnx(input, weight, bias, group, eps)
+    # FIXME: return fake value because we must return 3 outputs(norm, mean, var)
+    # We know how the 'mean' was computed in Torch, but don't know the 'var'
```

Comment:
why is it called `weight_inst_norm, bias_inst_norm` in `_aten_native_group_norm_onnx`, but called `fake_mean, fake_var` here?

Reply:
Looks like it;s here:
https://github.com/pytorch/pytorch/blob/8d3dc2131d6a866f826e4796c7402156aedf29b5/aten/src/ATen/native/group_norm.cpp#L64-L105

And
https://github.com/pytorch/pytorch/blob/main/aten/src/ATen/native/cpu/group_norm_kernel.cpp

Reply:
OK, I got the formula about Torch returned var (actually it is rstd) = ```1 / (sqrt( sum/n + eps))```, where ```n = HxW*C/N```